migration/migrator/migrations/course/20231206173634_course_materials.py

```diff
@@ -0,0 +1,42 @@
+"""Migration for a given Submitty course database."""
+
+
+def up(config, database, semester, course):
+    """
+    Run up migration.
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+    
+    # Add columns to the course_materials table.
+    database.execute("ALTER TABLE course_materials ADD COLUMN IF NOT EXISTS uploaded_by character varying(255) REFERENCES users(user_id) DEFAULT NULL;")
+    database.execute("ALTER TABLE course_materials ADD COLUMN IF NOT EXISTS upload_date TIMESTAMP WITH TIME ZONE DEFAULT NULL;")
+    database.execute("ALTER TABLE course_materials ADD COLUMN IF NOT EXISTS last_edit_by character varying(255) REFERENCES users(user_id) DEFAULT NULL;")
+    database.execute("ALTER TABLE course_materials ADD COLUMN IF NOT EXISTS last_edit_date TIMESTAMP WITH TIME ZONE DEFAULT NULL;")
+
+    database.execute("UPDATE course_materials SET last_edit_by = uploaded_by, last_edit_date = upload_date;")
```

Comment:
What if this up migration was already run once?  Wouldn't this overwrite any values?

Reply:
It will overwrite the values and to prevent this I converted this into a single transaction. So, if the first column already exists then the update command will not run. 

I have done this in the latest commit.

Reply:
I believe the code does the exact same thing still.  The `IF EXISTS` has no effect on anything other than the `ALTER TABLE` statement.  Everything in a single migration is run in a single transaction, but that doesn't' really matter here.  A better solution is to set the value to the `upload_date` value if the column is null.

Reply:
Okay. I will update it when I will push the changes related to it when I will complete this PR.